click_option_group/_core.py

```diff
@@ -285,6 +302,14 @@ def handle_parse_result(self, option: GroupedOption, ctx: click.Context, opts: d
 
             raise click.UsageError(error_text, ctx=ctx)
 
+    def get_help_record(self, ctx: click.Context) -> ty.Tuple[str, str]:
+        for option_name, option in self.get_options(ctx).items():
+            if option.hidden:
+                error_text = f'Hidden options ("{option_name}") are not allowed inside RequiredAllOptionGroup.'
+                raise TypeError(error_text)
+
+        return super().get_help_record(ctx)
```

Comment:
Maybe we could add 'hidden' attribute to `forbidden_option_attrs` list instead?

Reply:
And I think we can add the additional info about option name here:
https://github.com/click-contrib/click-option-group/blob/a2e9d5843a117a3607ea758a18aa48b7782681b0/click_option_group/_core.py#L29-L32

Reply:
done. There's a slight downside that I first had to call the superclass's constructor before running the check, to get the name of the option.